### ISSUE TITLE
New platform issues

### ISSUE THREAD

**Issue by h-2** [seqan/seqan#2245]
:sob: 

https://buildd.debian.org/status/package.php?p=seqan2&suite=experimental

- [x] test_align_parallel_data_structures [ppc64el]
- [x] test_demo_tutorial_file_io_overview_solution3 [ppc64el, arm64, s390x, powerpc, ppc64]
- [x] test_demo_tutorial_sam_and_bam_io_example7 [ppc64el, arm64, s390x, powerpc, ppc64]
-  ~~enable UME simd in nightly builds on larix.~~

@rrahn can you look at the first one, I will have a look at the other two.

**Comment by rrahn** [comment]
Not tonight but maybe tomorrow. Got some other issues with some change in simd related code. :unamused:

**Comment by h-2** [comment]
> Not tonight but maybe tomorrow. 

Yeah, me too, definetely not before sunday.

**Comment by h-2** [comment]
#2233 introduced errors 2 & 3, likely an issue of signed/unsigned char over/underflows. Possibly just using `0xFF` instead of `'\xFF'` will already fix this.

**Comment by rrahn** [comment]
@h-2 can you please include UME simd 0.8.1 in the nightlies on larix.
I will do it for mac

**Comment by h-2** [comment]
Wow, would you guess the result of this program:
```cpp
#include <iostream>
int main()
{
    std::cout << ((32 - '!') == 0xFF) << '\n';
    std::cout << ((32 - '!') == '\xff') << '\n';

    std::cout << unsigned(0xFF) << '\n';
    std::cout << unsigned('\xff') << '\n';
    std::cout << unsigned(char(0xFF)) << '\n';
    std::cout << unsigned(char('\xff')) << '\n';

    std::cout << (char(32 - '!') == 0xFF) << '\n';
    std::cout << (char(32 - '!') == '\xff') << '\n';
}
```

Once with just g++ and once with `-funsigned-char`?

<details> 
  <summary>signed char</summary>

```
0
1
255
4294967295
4294967295
4294967295
0
1
```
</details>

<details> 
  <summary>unsigned char</summary>

```
0
0
255
255
255
255
1
1
```
</details>

That in combination with `typeof(char + char) == int` can really ruin the day.

**Comment by h-2** [comment]
fixed by #2248 and #2249

**Comment by h-2** [comment]
@mr-c Can you give -rc1 a try? Hopefully all new platform issues are resolved now.

PS: Link to to Debian Med was added!

**Comment by h-2** [comment]
now we have:
```
/«BUILDDIR»/seqan2-2.4.0~rc1+dfsg/tests/align_parallel/test_align_wavefront_alignment_scheduler.h:264 Assertion failed : val == 50 was: 59 != 50

stack trace:
  0          [0x4fb184]  seqan::ClassTest::fail() + 0x34
  1          [0x5031c0]  void SEQAN_TEST_test_align_parallel_wavefront_alignment_scheduler_async_with_exception<true>() + 0xb00
  2          [0x4f62d0]  main + 0x140
  3        [0xf7591648]  /lib/powerpc-linux-gnu/libc.so.6(+0x21648)
  4        [0xf7591818]  __libc_start_main + 0xd8

test_align_parallel_wavefront_alignment_scheduler_async_with_exception FAILED
```

@rrahn ? I vote for leaving this for 2.4.1 as I really want to release tomorrow.

**Comment by h-2** [comment]
@mr-c I have pushed a new -rc, can you trigger an update? It should include fixes for alpha.

**Comment by mr-c** [comment]
@h-2 Sure. From the Debian perspective `alpha` is an unofficial / less supported architecture.

**Comment by h-2** [comment]
now there are some new issues again, one on `powerpc` and one on `armel`. I really don't understand why these pop up and disappear every random build, but they can't be fixed for 2.4.0 anymore.

Moving these to 2.4.1 and assigning to @rrahn.

**Comment by h-2** [comment]
@mr-c amd64 is now failing because of timeouts. Maybe the result of your reduction in threads?

**Comment by mr-c** [comment]
@h-2 Yep, already submitted :-) https://tracker.debian.org/news/930829

**Comment by mr-c** [comment]
a hint about sparc64: 

> sparc64 is notoriously strict about memory access alignment.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892223

https://docs.oracle.com/cd/E19253-01/816-4854/hwovr-2/index.html

**Comment by rrahn** [comment]
Mhmmm, many of the failed tests seem to do some sort of I/O. Maybe this is related to reading binary data from disk?
@mr-c Are always the same tests failing or is it random?

**Comment by mr-c** [comment]
Seems that SIGBUS is about memory access alignment errors, not IO. My feeling is that it is the same tests that fail, but I haven't looked.

**Comment by h-2** [comment]
We have limited reasources for Seqan2 now, is anything here really important to get done?
